core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -781,26 +785,31 @@ private Function createFunction(
                 }
             } else if ((argTypeTag == ColumnType.STRING || argTypeTag == ColumnType.SYMBOL) && sigArgTypeTag == ColumnType.TIMESTAMP) {
                 // convert arguments if necessary
```

Comment:
Why not push `assert arg.isConstant()` to if condition here and below  ? 
It could `explode` with AssertionError  if assertions are enabled.

Reply:
That's actually something I wanted to ask you: Is it possible that arg is ever not a constant? 

This is my understanding: The overloading rules as defined in `ColumnType` do not allow matching a function declared as receiving `Date` with a String argument. So a function which declares an argument of a type `Date` will match with an actual String argument if and only if the string is a constant. Only because of the code/hack at the lines 679-682. An that applies to constants only. No other case is allowed. And the assertion is meant to communicate this invariant. Does it make sense?

Reply:
You're right, non-const case is not reachable, so I guess it's fine .